### ISSUE TITLE
hook ActionOrderGridQueryBuilderModifier returns incorrect order count

### ISSUE THREAD

**Issue by musaffar-patel** [PrestaShop/PrestaShop#26858]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hi

When using the hook "actionOrderGridQueryBuilderModifier" one can use the searchQueryBuilder to modify the query which will return the list of orders in the Order list page in the back office.

The above works as expected, I can add conditions and the list of orders correctly corresponds to the list of conditions.  However the problem is that the number of orders (order count) and the pagination does not correctly reflect the actual number of orders displayed.

Attaching a screen shot to highlight the issue.

Kind Regards
Musaffar

![screenshot](https://user-images.githubusercontent.com/44204668/144607998-a6faeb80-5ea4-4c3a-94c2-5215d63cf3de.jpg)

### Steps to reproduce

Create a new module
Implement the hookActionOrderGridQueryBuilderModifier and hookActionOrderGridDefinitionModifier hook
in the hook add conditions to the $searchQueryBuilder object which would filter orders by a specific condition.
Now open the order list in the back office and filter by the new column implemented by the module.

### PrestaShop version(s) where the bug happened

1.7.7 to 1.7.8.1

**Comment by hibatallahAouadni** [comment]
Hello @musaffar-patel 

Could you please check this issue #26523 seems similar to yours.
Please check and feedback.

Thanks!

**Comment by musaffar-patel** [comment]
> Hello @musaffar-patel
> 
> Could you please check this issue #26523 seems similar to yours. Please check and feedback.
> 
> Thanks!

Hi, I've reviewed the issue you mentioned and it does not seem like the same issue.

In my case the new column I added is a new date field (related to my module).  I can enter dates and the orders are filtered correctly via the hook ActionOrderGridQueryBuilderModifier

So far no problems.

However, Prestashop does not display the correct order count.  If the hook modifies the query to return fewer results, then fewer results are displayed but the order count doe snot reflect the filtered count.
